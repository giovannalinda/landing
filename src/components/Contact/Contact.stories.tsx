import { Meta, Story } from '@storybook/react'

import { Contact } from './Contact'
import { ContactInfo } from './ContactInfo'
import { ContactForm } from './ContactForm'

import * as S from './Contact.styled'

export default {
  title: 'Components/Contact',
  component: Contact,
} as Meta

export const Form: Story = () => (
  <S.Container>
    <S.FormContainer>
      <ContactForm />
    </S.FormContainer>
  </S.Container>
)

export const Info: Story = () => (
  <S.Container>
    <S.InfoContainer>
      <ContactInfo />
    </S.InfoContainer>
  </S.Container>
)
