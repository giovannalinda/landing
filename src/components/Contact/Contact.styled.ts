import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.section`
  display: grid;
  grid-template-columns: 1.3fr 1fr;
  padding-top: 12.7rem;

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    grid-template-columns: 1fr;
  }

  h1 {
    padding-bottom: ${theme.space[5]};
    font-size: ${theme.font.sizes.subtitle};
    font-weight: ${theme.font.weigths.regular};
  }
`

export const FormContainer = styled.div`
  background: ${theme.colors.white};
  color: ${theme.colors.primaryDark};
  padding: ${theme.space[8]};
  border-radius: ${theme.borderRadius.lg};
`

export const InfoContainer = styled.div`
  position: relative;
  background: ${theme.colors.white};
  color: ${theme.colors.primaryDark};
  padding: ${theme.space[8]};
  border-radius: ${theme.borderRadius.lg};
  margin-left: ${theme.space[5]};

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    margin: ${theme.space[2]} 0 0 0;
    padding: ${theme.space[5]};
  }
`

export const DotImageWrapper = styled.div`
  position: absolute;
  right: -8rem;
  top: -3rem;

  @media screen and (max-width: ${theme.breakpoints.mobile}) {
    display: none;
  }
`
