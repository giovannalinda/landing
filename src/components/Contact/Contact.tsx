import Image from 'next/image'
import { Assets } from '~/assets'

import { Info } from './Info'
import { ContactForm } from './ContactForm'

import * as S from './Contact.styled'

export function Contact() {
  return (
    <S.Container id='contact'>
      <S.ContactForm>
        <h1>Let me know here.</h1>

        <ContactForm />
      </S.ContactForm>

      <S.ContactInfo>
        <S.ContainerImage>
          <Image src={Assets.Dot} width='152' height='68' />
        </S.ContainerImage>
        <Info />
      </S.ContactInfo>
    </S.Container>
  )
}
