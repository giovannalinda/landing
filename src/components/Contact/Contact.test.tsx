import { render } from '~/__helpers__/app-tests'
import { Contact } from './Contact'

describe('Contact', () => {
  it('should match snapshot', () => {
    const { container } = render(<Contact />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
