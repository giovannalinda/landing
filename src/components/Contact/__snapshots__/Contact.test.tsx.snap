// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should match snapshot 1`] = `
<section
  class="sc-kgUAyh eYVvfQ"
  id="contact"
>
  <div
    class="sc-hTtwUo kQTmMn"
  >
    <h1>
      Let me know here.
    </h1>
    <form
      class="sc-jOrMOR dyzBkV"
    >
      <div
        class="sc-dPyBCJ hlYoLC"
      >
        <input
          class="sc-bBXxYQ ChFDK"
          placeholder="Full name"
          type="text"
        />
        <input
          class="sc-bBXxYQ ChFDK"
          placeholder="Email adress"
          type="email"
        />
      </div>
      <input
        class="sc-bBXxYQ ChFDK"
        placeholder="Subjects"
        type="text"
      />
      <textarea
        class="sc-cOFTSb bcWpvL"
        placeholder="Message"
        rows="7"
      />
      <button
        class="sc-gsnTZi sc-BeQoi fBRaJm uFHTF"
        type="submit"
      >
        Send message
         
        <svg
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="22"
            x2="11"
            y1="2"
            y2="13"
          />
          <polygon
            points="22 2 15 22 11 13 2 9 22 2"
          />
        </svg>
      </button>
    </form>
  </div>
  <div
    class="sc-jgbSNz gMKhmQ"
  >
    <div
      class="sc-gSAPjG jFCfRO"
    >
      <img
        height="68"
        src="/dot.svg"
        width="152"
      />
    </div>
    <h1>
      Get In Touch
    </h1>
    <ul
      class="sc-iAvgwm ueAml"
    >
      <li>
        <div
          class="sc-efBctP gBpybl"
        >
          <svg
            fill="none"
            height="26"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"
            />
            <circle
              cx="12"
              cy="10"
              r="3"
            />
          </svg>
        </div>
        <small>
          Diadema,
          <br />
           SÃ£o Paulo - SP, Brazil.
        </small>
      </li>
      <li>
        <div
          class="sc-efBctP gBpybl"
        >
          <svg
            fill="none"
            height="26"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
            />
            <polyline
              points="22,6 12,13 2,6"
            />
          </svg>
        </div>
        <small>
          eugiovannasouza@gmail.com 
          <br />
          oigiovannalinda@gmail.com
        </small>
      </li>
      <li>
        <div
          class="sc-efBctP gBpybl"
        >
          <svg
            fill="none"
            height="26"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
            />
          </svg>
        </div>
        <small>
          +55 955 258 2699 
          <br />
           +55 955 100 9449
        </small>
      </li>
    </ul>
    <div
      class="sc-cTQhss jBWkJd"
    >
      <a
        href="https://dribbble.com/giovannalinda"
        target="_blank"
      >
        <button
          class="sc-gsnTZi YXSLR"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 16c-4.412 0-8-3.588-8-8s3.587-8 8-8c4.412 0 8 3.587 8 8s-3.588 8-8 8v0zM14.747 9.094c-0.234-0.075-2.116-0.634-4.256-0.291 0.894 2.456 1.256 4.456 1.328 4.872 1.531-1.037 2.625-2.678 2.928-4.581v0zM10.669 14.3c-0.103-0.6-0.497-2.688-1.456-5.181-0.016 0.006-0.031 0.009-0.044 0.016-3.856 1.344-5.241 4.016-5.362 4.266 1.159 0.903 2.616 1.444 4.194 1.444 0.947 0 1.85-0.194 2.669-0.544v0zM2.922 12.578c0.156-0.266 2.031-3.369 5.553-4.509 0.088-0.028 0.178-0.056 0.269-0.081-0.172-0.388-0.359-0.778-0.553-1.159-3.409 1.022-6.722 0.978-7.022 0.975-0.003 0.069-0.003 0.138-0.003 0.209 0 1.753 0.666 3.356 1.756 4.566v0zM1.313 6.609c0.306 0.003 3.122 0.016 6.319-0.831-1.131-2.013-2.353-3.706-2.534-3.953-1.913 0.903-3.344 2.666-3.784 4.784v0zM6.4 1.366c0.188 0.253 1.431 1.944 2.55 4 2.431-0.909 3.459-2.294 3.581-2.469-1.206-1.072-2.794-1.722-4.531-1.722-0.55 0.003-1.088 0.069-1.6 0.191v0zM13.291 3.691c-0.144 0.194-1.291 1.663-3.816 2.694 0.159 0.325 0.313 0.656 0.453 0.991 0.050 0.119 0.1 0.234 0.147 0.353 2.275-0.284 4.534 0.172 4.759 0.219-0.016-1.612-0.594-3.094-1.544-4.256v0z"
            />
          </svg>
        </button>
      </a>
      <a
        href="https://github.com/giovannalinda"
        target="_blank"
      >
        <button
          class="sc-gsnTZi YXSLR"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0.198c-4.418 0-8 3.582-8 8 0 3.535 2.292 6.533 5.471 7.591 0.4 0.074 0.547-0.174 0.547-0.385 0-0.191-0.008-0.821-0.011-1.489-2.226 0.484-2.695-0.944-2.695-0.944-0.364-0.925-0.888-1.171-0.888-1.171-0.726-0.497 0.055-0.486 0.055-0.486 0.803 0.056 1.226 0.824 1.226 0.824 0.714 1.223 1.872 0.869 2.328 0.665 0.072-0.517 0.279-0.87 0.508-1.070-1.777-0.202-3.645-0.888-3.645-3.954 0-0.873 0.313-1.587 0.824-2.147-0.083-0.202-0.357-1.015 0.077-2.117 0 0 0.672-0.215 2.201 0.82 0.638-0.177 1.322-0.266 2.002-0.269 0.68 0.003 1.365 0.092 2.004 0.269 1.527-1.035 2.198-0.82 2.198-0.82 0.435 1.102 0.162 1.916 0.079 2.117 0.513 0.56 0.823 1.274 0.823 2.147 0 3.073-1.872 3.749-3.653 3.947 0.287 0.248 0.543 0.735 0.543 1.481 0 1.070-0.009 1.932-0.009 2.195 0 0.213 0.144 0.462 0.55 0.384 3.177-1.059 5.466-4.057 5.466-7.59 0-4.418-3.582-8-8-8z"
            />
          </svg>
        </button>
      </a>
      <a
        href="https://www.linkedin.com/in/giovannalinda"
        target="_blank"
      >
        <button
          class="sc-gsnTZi YXSLR"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            version="1.1"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 6h2.767v1.418h0.040c0.385-0.691 1.327-1.418 2.732-1.418 2.921 0 3.461 1.818 3.461 4.183v4.817h-2.885v-4.27c0-1.018-0.021-2.329-1.5-2.329-1.502 0-1.732 1.109-1.732 2.255v4.344h-2.883v-9z"
            />
            <path
              d="M1 6h3v9h-3v-9z"
            />
            <path
              d="M4 3.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5c0-0.828 0.672-1.5 1.5-1.5s1.5 0.672 1.5 1.5z"
            />
          </svg>
        </button>
      </a>
      <a
        href="https://www.instagram.com/_gripada"
        target="_blank"
      >
        <button
          class="sc-gsnTZi YXSLR"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0C5.829 0 5.556.01 4.703.048 3.85.088 3.269.222 2.76.42a3.917 3.917 0 0 0-1.417.923A3.927 3.927 0 0 0 .42 2.76C.222 3.268.087 3.85.048 4.7.01 5.555 0 5.827 0 8.001c0 2.172.01 2.444.048 3.297.04.852.174 1.433.372 1.942.205.526.478.972.923 1.417.444.445.89.719 1.416.923.51.198 1.09.333 1.942.372C5.555 15.99 5.827 16 8 16s2.444-.01 3.298-.048c.851-.04 1.434-.174 1.943-.372a3.916 3.916 0 0 0 1.416-.923c.445-.445.718-.891.923-1.417.197-.509.332-1.09.372-1.942C15.99 10.445 16 10.173 16 8s-.01-2.445-.048-3.299c-.04-.851-.175-1.433-.372-1.941a3.926 3.926 0 0 0-.923-1.417A3.911 3.911 0 0 0 13.24.42c-.51-.198-1.092-.333-1.943-.372C10.443.01 10.172 0 7.998 0h.003zm-.717 1.442h.718c2.136 0 2.389.007 3.232.046.78.035 1.204.166 1.486.275.373.145.64.319.92.599.28.28.453.546.598.92.11.281.24.705.275 1.485.039.843.047 1.096.047 3.231s-.008 2.389-.047 3.232c-.035.78-.166 1.203-.275 1.485a2.47 2.47 0 0 1-.599.919c-.28.28-.546.453-.92.598-.28.11-.704.24-1.485.276-.843.038-1.096.047-3.232.047s-2.39-.009-3.233-.047c-.78-.036-1.203-.166-1.485-.276a2.478 2.478 0 0 1-.92-.598 2.48 2.48 0 0 1-.6-.92c-.109-.281-.24-.705-.275-1.485-.038-.843-.046-1.096-.046-3.233 0-2.136.008-2.388.046-3.231.036-.78.166-1.204.276-1.486.145-.373.319-.64.599-.92.28-.28.546-.453.92-.598.282-.11.705-.24 1.485-.276.738-.034 1.024-.044 2.515-.045v.002zm4.988 1.328a.96.96 0 1 0 0 1.92.96.96 0 0 0 0-1.92zm-4.27 1.122a4.109 4.109 0 1 0 0 8.217 4.109 4.109 0 0 0 0-8.217zm0 1.441a2.667 2.667 0 1 1 0 5.334 2.667 2.667 0 0 1 0-5.334z"
            />
          </svg>
        </button>
      </a>
      <a
        href="https://medium.com/@eugiovannasouza"
        target="_blank"
      >
        <button
          class="sc-gsnTZi YXSLR"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.025 8c0 2.485-2.02 4.5-4.513 4.5A4.506 4.506 0 0 1 0 8c0-2.486 2.02-4.5 4.512-4.5A4.506 4.506 0 0 1 9.025 8zm4.95 0c0 2.34-1.01 4.236-2.256 4.236-1.246 0-2.256-1.897-2.256-4.236 0-2.34 1.01-4.236 2.256-4.236 1.246 0 2.256 1.897 2.256 4.236zM16 8c0 2.096-.355 3.795-.794 3.795-.438 0-.793-1.7-.793-3.795 0-2.096.355-3.795.794-3.795.438 0 .793 1.699.793 3.795z"
            />
          </svg>
        </button>
      </a>
    </div>
  </div>
</section>
`;
